omnibus/cookbooks/omnibus-supermarket/libraries/config.rb

```diff
@@ -64,12 +64,20 @@ def self.audit_config(config)
     end
 
     def self.audit_s3_config(config)
-      required_s3_vars = %w(s3_bucket s3_access_key_id s3_secret_access_key s3_region).freeze
-      any_s3_settings = required_s3_vars.any? { |key| !(config[key].nil? || config[key].empty?) }
-      all_s3_settings = required_s3_vars.all? { |key| !(config[key].nil? || config[key].empty?) }
+      required_s3_vars = %w(s3_bucket s3_region).freeze
+      any_required_s3_vars = required_s3_vars.any? { |key| config.has_key?(key) }
```

Comment:
Found a bug my old check! When everything is set to empty strings, no exception would raise! This fixes that and there are some regression tests to confirm it.

Reply:
I am now suspicious of _this_ fix. The chefspecs for this cookbook very much to not like this.